refactor(storage): ensure output dir is cleared before each run

Koech01 opened the pull request:
### Summary

This PR ensures that the output directory is cleared before each notebook run. This prevents stale or leftover files from previous executions from affecting new runs.

### Changes Made

- Added a new method `clear_output_dir()` to `StorageManager` to delete and recreate the worker's output directory
- Called `clear_output_dir()` at the beginning of `run_notebook()` before syncing output 

### Related Issue

Fixes #333

### Notes

- Manually tested using a custom `.ipynb` notebook file

Koech01 committed:
refactor(storage): ensure output dir is cleared before each run

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/mljar/mercury?pullRequest=493) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/mljar/mercury?pullRequest=493) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/mljar/mercury?pullRequest=493) it.</sub>

Koech01 committed:
Moved clear_output_dir() call into run_notebook(self, json_params) to ensure output directory is cleaned before every execution

- Prevents stale or leftover files from affecting subsequent notebook runs
- Safely accessed notebook.task_id using getattr to avoid AttributeError

Fixes #333

pplonski commented:
Thank you! We have released a new version of Mercury in which we directly use `ipywidgets`. We use Tornado (the same server as Jupyter) for serving Mercury apps.
